substrate/frame/merkle-mountain-range/src/mmr/storage.rs

```diff
@@ -47,6 +46,26 @@ pub struct RuntimeStorage;
 /// DOES NOT support adding new items to the MMR.
 pub struct OffchainStorage;
 
+impl OffchainStorage {
+	fn get(key: &[u8]) -> Option<Vec<u8>> {
+		sp_io::offchain::local_storage_get(StorageKind::PERSISTENT, &key)
+	}
+
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	fn set(key: &[u8], value: &[u8]) {
+		sp_io::offchain_index::set(key, value);
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn set(key: &[u8], value: &[u8]) {
+		if sp_io::offchain::is_offchain_db_ext_available() {
+			sp_io::offchain::local_storage_set(StorageKind::PERSISTENT, key, value);
+		} else {
+			sp_io::offchain_index::set(key, value);
```

Comment:
Doesn't `sp_io::offchain_index::set(key, value);` check if extension is available internally?

Reply:
AFAIU `sp_io::offchain_index::set(key, value);` writes to a buffer, which is commited to offchain storage if the transaction succeeds and if the offchain_db ext is available.

The problem is that when running benchmarks, the values are not commited to offchain storage. But good point, I'll check why.

Reply:
I think it's because the benchmark initialization code is outside of the extrinsic, and the changes that happen there are not commited anywhere. Anyway, rolled back https://github.com/paritytech/polkadot-sdk/commit/9fab84aa2c3eb72395821d77e9804c576dd117ee and added a helper storage variable, only for `runtime-benchmarks` instead.